File: Programs/01-Lumberjack-Pile-Problem/solutions/askl56-lumberjack-pile-problem.py
n, to_place, *piles = [int(n) for n in open("input.txt").read().split()]

for i in range(to_place):
    piles[piles.index(min(piles))] += 1

for i in range(0, n*n, n):
    print(*piles[i:i+n])

File: Programs/01-Lumberjack-Pile-Problem/solutions/askl56-lumberjack-pile-problem.rb
n = gets.to_i
logs = gets.to_i

piles = Array.new(n) { gets.split.map &:to_i }.flatten

logs.times { piles[piles.index(piles.min)] += 1 }

piles.each_slice(n) { |row| puts row.join ' ' }
